### PR TITLE
Fix file path forgotten after 2019-->2021 migration (XML geometry) 

### PR DESCRIPTION
**Pull request by boudoul**
as mentioned in this Issue : #26771 (thanks @christopheralanwest , and also @abdoulline  ,  @mazarkin for noticing ) there was a mistake introdcued after the 2019-->2021 migration , a fileptah was incorrect - This PR is fixing this issue. 

#### PR validation:
It should not affect any std WF (where the geometry is read via the DB) , but I ran few events calling explicitely the 2021 XML : 
runTheMatrix.py -l 11623 --command '--geometry Extended2021'

### CODE REVIEW

**Comment by boudoul** [comment]
type bugfix

File: Geometry/CMSCommonData/python/cmsExtendedGeometry2021XML_cfi.py [hunk 1]
```diff
@@ -275,7 +275,7 @@
         'Geometry/EcalSimData/data/ESProdCuts.xml',
         'Geometry/TrackerSimData/data/PhaseI/trackerProdCuts.xml',
         'Geometry/TrackerSimData/data/trackerProdCutsBEAM.xml',
-        'Geometry/MuonSimData/data/2021/muonProdCuts.xml',
+        'Geometry/MuonSimData/data/2021/v1/muonProdCuts.xml',
```
Comment (ianna) [inline Geometry/CMSCommonData/python/cmsExtendedGeometry2021XML_cfi.py]: @boudoul - shouldn't be ```Geometry/MuonSimData/data/muonProdCuts/2021/v1/muonProdCuts.xml```?

**Review by ianna — CHANGES_REQUESTED** [review]

File: Geometry/CMSCommonData/python/cmsExtendedGeometry2021XML_cfi.py (hunk 1 above)
Comment (ianna) [reply Geometry/CMSCommonData/python/cmsExtendedGeometry2021XML_cfi.py]: Please, see rule No5 in https://github.com/cms-sw/cms-sw.github.io/blob/master/cms_coding_rules.md

Comment (boudoul) [reply Geometry/CMSCommonData/python/cmsExtendedGeometry2021XML_cfi.py]: hi @ianna .. mmh you are suggesting to add yet another subdirectory to https://github.com/cms-sw/cmssw/tree/master/Geometry/MuonSimData/data  ?   Geometry/MuonSimData/data/muonProdCuts doesn't exist ...

Comment (ianna) [reply Geometry/CMSCommonData/python/cmsExtendedGeometry2021XML_cfi.py]: sorry, it's not me, the coding rules do :-)

**Comment by ianna** [comment]
-1

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
+1

**Comment by kpedro88** [comment]
+upgrade

**Comment by fabiocos** [comment]
+1

**Pull request merged**